src/community/oseo/oseo-stac/src/main/java/org/geoserver/ogcapi/v1/stac/STACSearchQuery.java

```diff
@@ -46,4 +58,37 @@ public Integer getStartIndexAsInt() {
         }
         return startIndex.intValue();
     }
+
+    @Override
+    public void setSortBy(JsonNode node) {
+        // Based on STAC API Extension for Sort
+        // https://github.com/stac-api-extensions/sort?tab=readme-ov-file#http-post-json-entity
+        if (node instanceof ArrayNode) {
+            ArrayNode arrayNode = (ArrayNode) node;
+            SortBy[] sortBIES = new SortBy[arrayNode.size()];
+            for (int i = 0; i < arrayNode.size(); i++) {
+                if (arrayNode.get(i).isObject()
+                        && arrayNode.get(i).has(STAC_SORTBY_FIELD)
+                        && arrayNode.get(i).has(STAC_SORTBY_DIRECTION)) {
```

Comment:
I find having a mandatory direction odd, and yet the schema in the spec has it as mandatory, indeed.
[I've asked for a clarification](https://github.com/stac-api-extensions/sort/issues/10), in the meantime, this is correct.